https-repo.sh

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Martin Pitt <mpitt@redhat.com>
+
+TESTTYPE="method"
+
+. ${KSTESTDIR}/functions.sh
+
+prepare() {
+    ks=$1
+    tmpdir=$2
+    sed -e 's|http://\(.*fedoraproject.org\)|https://\1|' ${ks} > ${tmpdir}/kickstart.ks
```

Comment:
I would prefer to include a different fragment in the kickstart file. Please, create a new file `fragments/platform/fedora_rawhide/repos/secure.ks` with the old content of `default.ks`. Then you can replace `%ksappend repos/default.ks` with `%ksappend repos/secure.ks`.

Reply:
Good idea, this is less hacky. Done, PTAL?